app/webpacker/components/ContactEditProfilePage/index.jsx

```diff
@@ -45,6 +36,39 @@ export default function ContactEditProfilePage({ loggedInUserId, recaptchaPublic
     );
   }
 
+  if (queryParams.editOthersProfile) {
+    return (
+      <ContactEditProfileOthers
+        setContactSuccess={setContactSuccess}
+        recaptchaPublicKey={recaptchaPublicKey}
+      />
+    );
+  }
+  return (
+    <ContactEditProfileSelf
+      setContactSuccess={setContactSuccess}
+      recaptchaPublicKey={recaptchaPublicKey}
+    />
+  );
+}
+
+function ContactEditProfileSelf({ setContactSuccess, recaptchaPublicKey }) {
+  const { data: loggedInUserData, isLoading, isError } = useQuery({
+    queryKey: ['userData'],
+    queryFn: () => fetchJsonOrError(apiV0Urls.users.me.userDetails),
+  }, CONTACT_EDIT_PROFILE_QUERY_CLIENT);
+  const wcaId = loggedInUserData?.data?.user?.wca_id;
```

Comment:
Am I correct that you're running this whole query just to get the current user's WCA ID?

Reply:
Yes, in future maybe we can have a hook like `useLoggedInUserPermission` to get the WCA ID to make it simpler.